test/sys/test_signalfd.rs

```diff
@@ -0,0 +1,52 @@
+#[cfg(target_os = "linux")]
+#[test]
+// FIXME: Fails with "fatal runtime error: failed to initiate panic, error 5" on i686-unknown-linux-musl
+#[cfg_attr(all(target_env = "musl", target_arch = "x86"), ignore)]
+fn test_signalfd() {
+    use nix::sys::signalfd::{SFD_NONBLOCK, SignalFd};
+    use nix::sys::signal;
+    use nix::unistd;
+    use nix::unistd::*;
+    use nix::unistd::ForkResult::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::*;
+
+    // Set up a signalfd to listen for SIGUSR1 on this thread
+    let mut mask = SigSet::empty();
+    mask.add(SIGUSR1);
+    mask.thread_block().unwrap();
+
+    // Start a child process
+    match unistd::fork() {
```

Comment:
I remember why I was doing this, because if I didn't I saw errors like "process didn't exit successfully (signal: 10)" even just running my test by itself. This is `ECHILD`, which states there are no child processes. Would you have another suggestion rather than `fork`ing?